### PR TITLE
workflows: linux: Compile and execute internal test on non-amd64 platforms Linux

### PR DESCRIPTION
**Pull request by cosmo0920**
This workflow could be used for aarch64 Linux testing purpose such as #8851.

Also, `uraimo/run-on-arch-action@v2` workflow component can handle s390x and ppc64le architectures by QEMU emulations.
So, we can add these targets in the future.

----
Enter `[N/A]` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [ ] Example configuration file for the change
- [ ] Debug log output from testing the change

- [ ] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [ ] Documentation required for this feature

**Backporting**

- [ ] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

### CODE REVIEW

File: .github/workflows/unit-tests.yaml [hunk 1]
```diff
@@ -110,6 +110,52 @@ jobs:
           CXX: g++
           FLB_OPT: ${{ matrix.flb_option }}
 
+
+  run-qemu-aarch64-unit-tests:
+    # We chain this after Linux one as there are CPU time costs for QEMU emulation
+    needs:
+      - run-ubuntu-unit-tests
+    runs-on: ubuntu-20.04
```
Comment (patrick-stephens) [inline .github/workflows/unit-tests.yaml:118]: We can run on the ARM runners from CNCF using Actuated btw, we use them for building already and other projects use for testing

Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml:118]: Got it. I found the actuated set up line: https://github.com/fluent/fluent-bit/blob/master/.github/workflows/call-build-linux-packages.yaml#L125

The main point of using this emulated workflow is not required for budget. Plus, this workflow can be extended to use for s390x and ppc64le(PowerPC 64bit Little Endian).
We really want to confirm these platforms in the future.

Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml:118]: In addition, the emulated aarch64 builds are spent less than 30 minutes. It's actually less than time costs of coverage test (~50 mins)

Comment (leonardo-albertovich) [reply .github/workflows/unit-tests.yaml:118]: I think we should leave the scaffolding in place to use it for s390 as you mentioned but we should use actuated for aarch64 as Pat said.

Comment (patrick-stephens) [reply .github/workflows/unit-tests.yaml:175]: Budget-wise I think we're fine but keeping it simple is better I'd say so all good. Main thing was to ensure we don't block for some really long time - like the builds do if emulated!

Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml:175]: For blocking build perspective, it won't be a blocker even if using emulated build. On of the main candidates of the blockers is coverage task which is required for over 40 minutes long.

Comment (patrick-stephens) [reply .github/workflows/unit-tests.yaml:175]: Yup, personally I'm always preferring tests we can run locally as well so emulation is nice for that.

**Comment by cosmo0920** [comment]
The aarch64 unit testing on actuated test runner got green: https://github.com/fluent/fluent-bit/actions/runs/10700447782

File: .github/workflows/unit-tests.yaml [hunk 2]
```diff
@@ -110,6 +110,115 @@ jobs:
...
+    steps:
+      - name: Checkout Fluent Bit code
+        uses: actions/checkout@v4
+
+      - name: Set up Actuated mirror
+        if: github.repository == 'fluent/fluent-bit'
+        uses: self-actuated/hub-mirror@master
+
+      - name: Setup environment
+        if: github.repository == 'fluent/fluent-bit'
```
Comment (patrick-stephens) [inline .github/workflows/unit-tests.yaml:141]: I don't think we need this `if` as we can do it regardless, we only have it for the mirror step above when we're using the Actuated runner

File: .github/workflows/unit-tests.yaml [hunk 3]
```diff
@@ -110,6 +110,115 @@ jobs:
...
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gcc-9 g++-9 clang-12 cmake flex bison libsystemd-dev gcovr libyaml-dev
+          sudo ln -s /usr/bin/llvm-symbolizer-12 /usr/bin/llvm-symbolizer || true
+
+      - name: Build and test with actuated runners
+        if: github.repository == 'fluent/fluent-bit'
```
Comment (patrick-stephens) [inline .github/workflows/unit-tests.yaml:148]: No need for this, we want to run the tests regardless - if not then why are we running the job at all?

File: .github/workflows/unit-tests.yaml (hunk 2 above)
Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml]: I removed this condition. Without actuated runners, this workflow is just running as an ordinary amd64 worker. But that's fine.

File: .github/workflows/unit-tests.yaml (hunk 3 above)
Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml]: Fair enough. I removed this condition here.

File: .github/workflows/unit-tests.yaml [hunk 4]
```diff
@@ -110,6 +110,113 @@ jobs:
...
 
+  run-aarch64-unit-tests:
+    # Ensure for OSS Fluent Bit repo we enable usage of Actuated runners for ARM builds, for forks it should keep existing ubuntu-latest usage.
+    runs-on: ${{(github.repository == 'fluent/fluent-bit') && 'actuated-arm64-8cpu-16gb' || 'ubuntu-latest' }}
+    permissions:
+      contents: read
+    # We chain this after Linux one as there are costs for actuated workers
+    needs:
+      - run-ubuntu-unit-tests
+    timeout-minutes: 60
```
Comment (patrick-stephens) [inline .github/workflows/unit-tests.yaml:121]: Should we drop this? Seems very long - what should the upper bound be?

**Review by patrick-stephens — APPROVED** [review]
Looks fine from a CI perspective.

File: .github/workflows/unit-tests.yaml (hunk 4 above)
Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml:121]: Ah, good catch. Probably, we need to set up as 10.

Comment (cosmo0920) [reply .github/workflows/unit-tests.yaml]: I used 10mins for the actuated runners.

File: .github/workflows/unit-tests.yaml [hunk 5]
```diff
@@ -110,6 +110,113 @@ jobs:
...
+        working-directory: build
+        env:
+          CC: ${{ matrix.config.compiler }}
+          CXX: ${{ matrix.config.compiler }}
+
+  run-qemu-ubuntu-unit-tests:
+    # We chain this after Linux one as there are CPU time costs for QEMU emulation
+    needs:
+      - run-ubuntu-unit-tests
+    runs-on: ubuntu-20.04
```
Comment (patrick-stephens) [inline .github/workflows/unit-tests.yaml:173]: We probably want to look at stepping this up as well soon, 22 is the default and 24 is coming. Not a blocker now though.

**Pull request merged**